lib/async/http/faraday/adapter.rb

```diff
@@ -99,6 +99,9 @@ def close
 				def call(env)
 					super
 					
+					# for compatibility with the default adapter
+					env.url.path = '/' if env.url.path.empty?
```

Comment:
Is it safe to mutate the `env.url` input in this way? Is this how the normal Faraday adapter works? i.e. `env.url.path = ...`? Or should we dup the url and modify that?

Reply:
Normal faraday works with URLs that have no trailing slash such as "https://www.google.com" even if the client is created with no url as like as `Faraday.new()`
It means the Faraday expects the `path` component must not be empty.
I think the `env` argument is expected `mutable` among the middleware stack and the Faraday expects this URL normalization is due to be done by the adapter.
To be honest, my preference is the normalization is done by the Faraday itself, but the real isn't and fixing all adapters is hard work.